### PR TITLE
[#97] Supporting backup in AWS S3 engine

### PR DESCRIPTION
**Pull request by MariamFahmy98**
Implementing S3 storage engine for backups.

### CODE REVIEW

**Comment by jesperpedersen** [comment]
Lets get the documentation in as well.

* README.md (Storage engines (local, ssh, s3) (under * Restore))
* doc/CONFIGURATION.md
* doc/S3.md
* doc/man/pgmoneta.conf.5.rst

File: doc/CONFIGURATION.md [hunk 1]
```diff
@@ -32,12 +32,17 @@ See a [sample](./etc/pgmoneta.conf) configuration for running `pgmoneta` on `loc
...
 | ssh_hostname | | String | Yes | Defines the hostname of the remote system for connection |
 | ssh_username | | String | Yes | Defines the username of the remote system for connection |
 | ssh_base_dir | | String | Yes | The base directory for the remote backup |
 | ssh_ciphers | aes-256-ctr, aes-192-ctr, aes-128-ctr | String | No | The supported ciphers for communication. `aes \| aes-256 \| aes-256-cbc`: AES CBC (Cipher Block Chaining) mode with 256 bit key length<br/> `aes-192 \| aes-192-cbc`: AES CBC mode with 192 bit key length<br/> `aes-128 \| aes-128-cbc`: AES CBC mode with 128 bit key length<br/> `aes-256-ctr`: AES CTR (Counter) mode with 256 bit key length<br/> `aes-192-ctr`: AES CTR mode with 192 bit key length<br/> `aes-128-ctr`: AES CTR mode with 128 bit key length. Otherwise verbatim |
+| aws_region | | String | Yes | The AWS region |
+| access_key_id | | String | Yes | The IAM access key ID |
+| secret_access_key | | String | Yes | The IAM secret access key |
```
Comment (jesperpedersen) [inline doc/CONFIGURATION.md:43]: We should probably prefix these 3 with `s3_` as well...

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

File: doc/CONFIGURATION.md (hunk 1 above)
Comment (MariamFahmy98) [reply doc/CONFIGURATION.md]: Done

File: doc/man/pgmoneta.conf.5.rst [hunk 2]
```diff
@@ -106,6 +106,18 @@ ssh_ciphers
 
   Otherwise verbatim. Default is aes-256-ctr, aes-192-ctr, aes-128-ctr
 
+aws_region 
+  The AWS region 
+
+access_key_id
+  The IAM access key ID
+
+secret_access_key
```
Comment (jesperpedersen) [inline doc/man/pgmoneta.conf.5.rst:115]: Missing the `s3_` prefix here

**Review by jesperpedersen — CHANGES_REQUESTED** [review]

**Comment by jesperpedersen** [comment]
Merged.

Thanks for your contribution !

**Pull request merged**